lib/IRGen/GenRecord.h

```diff
@@ -47,7 +47,8 @@ template <class FieldImpl> class RecordField {
 
 protected:
   explicit RecordField(const TypeInfo &elementTI)
-    : Layout(ElementLayout::getIncomplete(elementTI)) {}
+      : Layout(ElementLayout::getIncomplete(elementTI)), Begin(0XFFF),
+        End(0XFFF) {}
```

Comment:
I think you meant `0xFFFF`.

Maybe it is better to leave `Begin` and `End` uninitialized so that MSan can catch a bogus read? Or you could add an `assert(Begin != 0xFFFF);` in `getProjectionRange()`.

Reply:
Fair enough. Whatever value's left on the stack will probably be a good enough indication anyway.